### PR TITLE
fix: `withImmer` doesn't work with resettable atom

### PR DESCRIPTION
**Pull request by tien**
Resolves pmndrs/jotai#2621

### CODE REVIEW

File: src/withImmer.ts [hunk 1]
```diff
@@ -7,14 +7,25 @@ const cache1 = new WeakMap();
...
+export function withImmer<Value, Args extends unknown[], Result>(
+  anAtom: WritableAtom<Value, Args, Result>,
+): WritableAtom<
+  Value,
+  Args extends [Value | infer OtherValue]
+    ? [
+        | Value
+        | ((draft: Draft<Value>) => void)
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        | Exclude<OtherValue, (...args: any[]) => unknown>,
```
Comment (dai-shi) [inline src/withImmer.ts:19]: Does `never` work here?
```suggestion
        | Exclude<OtherValue, (...args: never[]) => unknown>,
```

Comment (tien) [reply src/withImmer.ts]: Yeah, it does actually?? No idea how this make sense semantically, but I have been doing things wrong all this time then :)))

**Comment by tien** [comment]
> Oh, so it was just a type issue.
> 
> I'm not 100% sure how the order of overloads work, but it seems okay.

I believe it's like a switch case where the first matching pattern will be used, the original initial overload was too generic when I tested and ended up always shadowing the second overload.

**Comment by dai-shi** [comment]
I was curious if the second pattern would ever match.

**Review by dai-shi — APPROVED** [review]
The code looks good to me.
Thanks for your contribution!

**Comment by dai-shi** [comment]
Published: https://www.npmjs.com/package/jotai-immer/v/0.4.1

**Pull request merged**